### PR TITLE
implement vcard and use it for OAI-PMH-output

### PR DESCRIPTION
**Pull request by martinobersteiner**
`OAI-PMH` harvesters following the [LOM-UIBK spec](https://oer-repo.uibk.ac.at/lom/latest/) require the `lifecycle.contribute.centity.vcard` field to contain a `VCard`-formatted string (see [RFC6350](https://www.rfc-editor.org/rfc/rfc6350) for `VCard` spec).

Other packages don't seem to allow for easy creation of `VCard`s, so this PR includes a custom implementation for that.

### CODE REVIEW

File: invenio_records_lom/resources/serializers/schemas/oai.py [hunk 1]
```diff
@@ -164,7 +166,7 @@ def get_centity(self, entities):
 
         return [
             {
-                "vcard": e,  # TODO: fix vcard
+                "vcard": make_lom_vcard(fn=e),
```
Comment (utnapischtim) [inline invenio_records_lom/resources/serializers/schemas/oai.py:169]: minor: could this whole return statement written in one line? Maybe by removing the ',' after make_lom_vcard?

File: invenio_records_lom/utils/vcard.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+   vcard = my_vcard_maker.make_vcard(**supported_vcard_properties)
+   # `vcard`'s type will be type `str` or `bytes`, depending on `my_config`
+"""
+from __future__ import annotations  # allow `str | None` notation
+
+from collections.abc import Iterable
+from dataclasses import dataclass
+
+
+@dataclass
```
Comment (utnapischtim) [inline invenio_records_lom/utils/vcard.py:33]: question: Why do you use dataclass here? Is it only to not define the constructor for the 3 properties?

File: invenio_records_lom/utils/vcard.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+            header_lines + fn_lines + footer_lines
+        )
+        vcard = self.line_wrap(unwrapped_vcard)
+
+        if self.output_encoding:
+            return vcard.encode(encoding=self.output_encoding)
+        return vcard
+
+
+lom_vcard_maker = VCardMaker(
```
Comment (utnapischtim) [inline invenio_records_lom/utils/vcard.py:124]: opinion: I don't like constructs like that. What is wrong with defining a normal ordinary function? Like:

```
def make_lom_vcard(text: str) -> str | bytes:
    maker = VCardMaker(final_line_break=False, line_break_str="\n",output_encoding=None)
    return maker.make_vcard(text)
```

File: invenio_records_lom/utils/vcard.py [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+                else:
+                    idx += 1
+            # handle rest of `input_line`
+            output_lines.append(prefix + input_line[start:])
+
+        return self.line_break_str.join(output_lines) + (
+            self.line_break_str if self.final_line_break else ""
+        )
+
+    def make_vcard(self, fn: str | Iterable[str]) -> str | bytes:
```
Comment (utnapischtim) [inline invenio_records_lom/utils/vcard.py:98]: readability: what does `fn` stands for? I would interpret it as a function, but that makes no sense with the defined types

**Review by utnapischtim — COMMENTED** [review]
Where are the tests?

File: invenio_records_lom/utils/vcard.py (hunk 2 above)
Comment (martinobersteiner) [reply invenio_records_lom/utils/vcard.py:33]: `@dataclass` also adds some other dunder methods, which are recommended to be created anyway
(e.g. `__repr__`, `__eq__`, `__hash__`)

File: invenio_records_lom/utils/vcard.py (hunk 4 above)
Comment (utnapischtim) [reply invenio_records_lom/utils/vcard.py:98]: is `Names of arguments are the lower-cased version of properties defined by the spec.` defined by the spec or by you? I would prefer the written out way so fn -> formatted_name

File: invenio_records_lom/utils/vcard.py (hunk 2 above)
Comment (utnapischtim) [reply invenio_records_lom/utils/vcard.py:33]: yes, but why are they necessary in this context, if you only call `make_vcard` in the end?

**Review by utnapischtim — APPROVED** [review]

**Pull request merged**